Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -126,15 +153,59 @@ def draw(self, pen):
         how that works.
         """
         glyph, offset = self._getGlyphAndOffset()
+
+        if self.glyphSet.locationStack:
+            offset = 0  # Offset should only apply at top-level
+
+        if glyph.isVarComposite():
+            self._drawVarComposite(glyph, pen, False)
+            return
+
         glyph.draw(pen, self.glyphSet.glyfTable, offset)
 
     def drawPoints(self, pen):
         """Draw the glyph onto ``pen``. See fontTools.pens.pointPen for details
         how that works.
         """
         glyph, offset = self._getGlyphAndOffset()
+
+        if self.glyphSet.locationStack:
+            offset = 0  # Offset should only apply at top-level
+
+        if glyph.isVarComposite():
+            self._drawVarComposite(glyph, pen, True)
+            return
+
         glyph.drawPoints(pen, self.glyphSet.glyfTable, offset)
 
+    def _drawVarComposite(self, glyph, pen, isPointPen):
+
+        from fontTools.ttLib.tables._g_l_y_f import (
+            VarComponentFlags,
+            VAR_COMPONENT_TRANSFORM_MAPPING,
+        )
+
+        for comp in glyph.components:
+
+            with self.glyphSet.pushLocation(
+                comp.location, comp.flags & VarComponentFlags.RESET_UNSPECIFIED_AXES
+            ):
+                try:
+                    pen.addVarComponent(
+                        comp.glyphName, comp.transform, self.glyphSet.rawLocation
+                    )
+                except NotImplementedError:
```

Comment:
this could fail with AttributeError if a pen only implements the current interface (qucks like a pen) without actually subclassing directly from the fonttools base pens... is it intentional? or should we not instead decompose in that case? (see my other comment)

Reply:
Yeah, I just experienced this first-hand: my Fontra point pen does not inherit from AbstractPointPen (although it probably should).

That said: even flattening currently doesn't work for me like it did yesterday. Debugging.

Reply:
I pushed the fix. Hope you don't mind.